graphql_compiler/query_pagination/parameter_generator.py

```diff
@@ -1,4 +1,132 @@
 # Copyright 2019-present Kensho Technologies, LLC.
+import itertools
+import bisect
+
+from graphql.language.printer import print_ast
+
+from ..compiler.compiler_frontend import graphql_to_ir
+from ..compiler.helpers import Location
+from ..cost_estimation.interval import Interval, measure_int_interval, intersect_int_intervals
+from ..cost_estimation.helpers import is_int_field_type, is_uuid4_type
+from ..cost_estimation.filter_selectivity_utils import get_integer_interval_for_filters_on_field
+from ..cost_estimation.int_value_conversion import convert_field_value_to_int, convert_int_to_field_value, MIN_UUID_INT, MAX_UUID_INT, field_supports_range_reasoning
+
+
+def _get_query_path_endpoint_type(schema, query_path):
+    # TODO needs more care with:
+    # - non-null types
+    # - type coercions
+    current_type = schema.get_type('RootSchemaQuery')
+    for selection in query_path:
+        current_type = current_type.fields[selection].type.of_type
+    return current_type
+
+
+def _sum_partition(number, num_splits):
+    """Represent an integer as a sum of N almost equal integers, sorted in descending order.
+
+    Example: _sum_partition(5, 3) = [2, 2, 1]
+    """
+    lower = number // num_splits
+    num_high = number - lower * num_splits
+    num_low = num_splits - num_high
+    return itertools.accumulate(itertools.chain(
+        itertools.repeat(lower + 1, num_high),
+        itertools.repeat(lower, num_low - 1),
+    ))
+
+
+def _convert_int_interval_to_field_value_interval(schema_info, vertex_type, field, interval):
+    lower_bound = None
+    upper_bound = None
+    if interval.lower_bound is not None:
+        lower_bound = convert_int_to_field_value(
+            schema_info, vertex_type, field, interval.lower_bound)
+    if interval.upper_bound is not None:
+        upper_bound = convert_int_to_field_value(
+            schema_info, vertex_type, field, interval.upper_bound)
+    return Interval(lower_bound, upper_bound)
+
+
+def generate_parameters_for_vertex_partition(schema_info, query_ast, parameters, vertex_partition):
```

Comment:
It would probably be good to refactor and split up this function some more, there's kind of a lot going on there now. For example, there are at least two `return` statements, and the reader would have to check each of them for compliance with the generator's semantics.